packages/controller/src/lib/notificationHandler.ts

```diff
@@ -379,52 +477,55 @@ class NotificationHandler {
                 continue;
             }
 
-            res[scope] = { categories: {} };
-            res[scope].description = this.setup[scope].description;
-            res[scope].name = this.setup[scope].name;
+            res[scope] = { categories: {}, description: this.setup[scope].description, name: this.setup[scope].name };
 
             for (const category of Object.keys(this.currentNotifications[scope])) {
                 if (categoryFilter && categoryFilter !== category) {
                     // category filtered out
                     continue;
                 }
 
-                if (!this.setup[scope][category]) {
+                if (!this.setup[scope].categories[category]) {
                     // no category set up
                     continue;
                 }
 
-                res[scope].categories[category] = { instances: {} };
-                res[scope].categories[category].description = this.setup[scope][category].description;
-                res[scope].categories[category].name = this.setup[scope][category].name;
-                res[scope].categories[category].severity = this.setup[scope][category].severity;
+                res[scope]!.categories[category] = {
+                    instances: {},
+                    description: this.setup[scope].categories[category].description,
+                    name: this.setup[scope].categories[category].name,
+                    severity: this.setup[scope].categories[category].severity
```

Comment:
Please store `this.setup[scope].categories[category]` in a const at the start of the loop.